src/openms/include/OpenMS/KERNEL/MSExperiment.h

```diff
@@ -1067,16 +1070,64 @@ std::vector<MSChromatogram> extractXICs(
       @note The range values (min, max, etc.) are not updated automatically. Call updateRanges() to update the values!
     */
     ///@{
-    // Docu in base class
-    void updateRanges() override;
+    /// Delegate methods for backward compatibility
+    
+    /// Clear all ranges
+    void clearRanges() 
+    { 
+      combined_ranges_.clearRanges(); 
+      spectrum_ranges_.clearRanges();
+      chromatogram_ranges_.clearRanges();
+    }
+   
+    /// Get the minimum RT value of chromatogram and spectra ranges
+    double getMinRT() const { return combined_ranges_.getMinRT(); }
+    
+    /// Get the maximum RT value of chromatogram and spectra ranges
+    double getMaxRT() const { return combined_ranges_.getMaxRT(); }
+    
+    /// Get the minimum m/z value of chromatogram and spectra ranges
+    double getMinMZ() const { return combined_ranges_.getMinMZ(); }
+    
+    /// Get the maximum m/z value of chromatogram and spectra ranges
+    double getMaxMZ() const { return combined_ranges_.getMaxMZ(); }
+    
+    /// Get the minimum intensity value of chromatogram and spectra ranges
+    double getMinIntensity() const { return combined_ranges_.getMinIntensity(); }
+    
+    /// Get the maximum intensity value of chromatogram and spectra ranges
+    double getMaxIntensity() const { return combined_ranges_.getMaxIntensity(); }
+    
+    /// Get the minimum mobility value of chromatogram and spectra ranges
+    double getMinMobility() const { return combined_ranges_.getMinMobility(); }
+    
+    /// Get the maximum mobility value of chromatogram and spectra ranges
+    double getMaxMobility() const { return combined_ranges_.getMaxMobility(); }
+    
+    /// Extend the range to include the given RT value
+    void extendRT(double rt) { combined_ranges_.extendRT(rt); }
+    
+    /// Extend the range to include the given MZ value
+    void extendMZ(double mz) { combined_ranges_.extendMZ(mz); }
+    
+    /// Extend the range to include the given intensity value
+    void extendIntensity(double intensity) { combined_ranges_.extendIntensity(intensity); }
+    
+    /// Extend the range to include the given mobility value
+    void extendMobility(double mobility) { combined_ranges_.extendMobility(mobility); }
+    
+    /// Extend the range to include the given range
+    template<typename... RangeBasesOther>
+    void extend(const RangeManager<RangeBasesOther...>& rhs)
+    {
+      combined_ranges_.extendUnsafe(rhs);
+    }
 
     /**
-      @brief Updates the m/z, intensity, and retention time ranges of all spectra with a certain ms level
+      @brief Updates the m/z, intensity, mobility, and retention time ranges of all spectra and chromatograms
       
-
-      @param ms_level MS level to consider for m/z range, RT range and intensity range (All MS levels if negative)
     */
-    void updateRanges(Int ms_level);
```

Comment:
I removed the level type update mechanism.
I think it is not needed anymore with the separate ranges and performance should not be an issue.